kr8s/tests/test_api.py

```diff
@@ -314,6 +314,32 @@ async def test_api_timeout() -> None:
         await api.version()
 
 
+async def test_api_gone(example_pod_spec):
+    pod = await Pod(example_pod_spec)
+    await pod.create()
+
+    i = 0
+    with pytest.raises(GoneError):
+        # Start a watch request on the pod with a very small resourceVersion
+        async for _ in pod.api.async_watch(
+            pod.endpoint,
+            namespace=pod.namespace,
+            field_selector=f"metadata.name={pod.name}",
+            since=1,
+        ):
+            # Patch the pod to increment the resourceVersion if needed
+            await pod.patch(
+                {
+                    "metadata": {
+                        "annotations": {"version": str(i)},
+                    }
+                }
+            )
+            i += 1
```

Comment:
I think this is causing a deadlock because the patch creates and event, but we need an event to happen to call the patch code.

Instead this should probably be a task group with the watch and the patch happening in separate tasks.

Reply:
I rework the test to run the watch in a task and mock the `async_get_kind` response